docs/arch/plugins_actions.md

```diff
@@ -0,0 +1,53 @@
+# RDIFF-BACKUP ACTIONS PLUGINS
+
+## Naming convention and actions manager
+
+An action plug-in is:
+
+1. either in the `rdiffbackup.actions` namespace, with any module name (this is
+   how built-in action modules are packaged),
+2. or a top-level module with a name `rdb_action_[...].py`
+
+All those plug-ins are found by the action manager module `rdiffbackup.actions_mgr`
+and returned as a hash made of key-value pairs by the function `get_discovered_actions`:
+
+* key is the name returned by the Action class (see below the `get_name` method)
+* value is the Action class itself, as returned by the `get_action_class` function
+  of the action plug-in module.
+
+> **NOTE:** more information is provided with `pydoc rdiffbackup.actions_mgr`.
```

Comment:
I feel we are re-inventing the wheel. Should we consider using setuptools entry-point instead ? setup-tool entry point allow quick extension in a pure python environment with a standard approach.

It's very declarative too. 

If we use setup-tools, we may strip down the code you made. Meaning let code to maintain.

Reply:
My approach is actually coming directly from the Python packaging team, see https://packaging.python.org/guides/creating-and-discovering-plugins/, similar to the approach you suggest, listed as last option.

And let's be clear that the "entry point" is only the module itself; how the module internally looks like so that it can be used as plug-in still needs to be described/implemented.

Personally, I find that dropping a module named `rdb_action_whatever.py` in one's PYTHONPATH is a lot easier than any setup.py approach. We can still add the option if there is a demand later on (as said, it just defines how we find modules, not how we use them).

Reply:
I've raise the idea so we could consider using it. I don't mind keeping your implementation.